spec/requests/subjects_spec.rb

```diff
@@ -25,10 +25,10 @@
       expect(response.headers['Authorization']).to be_present
     end
 
-    describe 'GET /subjects as admin' do
+    describe 'GET /subjects as admin by giving a page number within total page number' do
       before do
         2.times { create_subject }
-        get listing_url, headers: headers
+        get listing_url, params: { page: 1 }, headers: headers
```

Comment:
@josephelias-rubyians  The code should take page 1 as default. So with your previous spec, the code should return the response correctly.

Reply:
@josephelias-rubyians  Write another spec for page 2 and test for the data.